api/src/main/java/org/openmrs/util/DeepCopyUtil.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.util;
+
+import java.util.Date;
+
+/**
+ * Utility class providing static methods supporting making deep copies of objects.  
+ */
+public class DeepCopyUtil {
+	
+	/**
+	 * Creates deep copy of a given String object
+	 * 
+	 * @param stringSrc string to be copied
+	 * @return deep copy of a given string
+	 */
+	public static String copy(String stringSrc) {
+		if (stringSrc == null) {
+			return null;
+		}
+		
+		return new String(stringSrc);
+	}
+	
+	/**
+	 * Creates deep copy of a given Date object
+	 * 
+	 * @param dateSrc Date to be copied
+	 * @return deep copy of a given Date
+	 */
+	public static Date copy(Date dateSrc) {
+		if (dateSrc == null) {
+			return null;
+		}
+		
+		return (Date) dateSrc.clone();
+	}
```

Comment:
@mjanuchowski you are right about deep copying the members like (PersonName, PersonAddress, PersonAttribute) 👍